avail/src/main/kotlin/avail/anvil/settings/TemplateExpansionsManager.kt

```diff
@@ -158,45 +170,287 @@ class RootTemplatesPanel constructor(
+					redrawTemplates()
+				}
+			}
+		}
+	}
+
 	init
 	{
 		layout = BoxLayout(this, BoxLayout.Y_AXIS)
+
+		val checkAllTemplate = JButton("Check All").apply {
+			isOpaque = true
+			val currentHeight = height
+			val currentWidth = width
+			minimumSize = Dimension(currentWidth + 140, currentHeight + 30)
+			preferredSize = Dimension(currentWidth + 140, currentHeight + 30)
+			maximumSize = Dimension(currentWidth + 140, currentHeight + 30)
+			toolTipText = "Create a new template"
+			addActionListener {
+				SwingUtilities.invokeLater {
+					redrawTemplates(true)
+					deleteSelectedTemplate.isEnabled = true
+				}
+			}
+		}
+		val uncheckAllTemplate = JButton("Uncheck All").apply {
+			isOpaque = true
+			val currentHeight = height
+			val currentWidth = width
+			minimumSize = Dimension(currentWidth + 140, currentHeight + 30)
+			preferredSize = Dimension(currentWidth + 140, currentHeight + 30)
+			maximumSize = Dimension(currentWidth + 140, currentHeight + 30)
+			toolTipText = "Create a new template"
+			addActionListener {
+				SwingUtilities.invokeLater {
+					redrawTemplates()
+					deleteSelectedTemplate.isEnabled = false
+				}
+			}
+		}
+
+		add(JPanel().apply {
+			layout = (FlowLayout(FlowLayout.LEFT))
+			minimumSize = Dimension(950, 50)
+			preferredSize = Dimension(950, 50)
+			maximumSize = Dimension(950, 50)
+			background = Color(0x3C, 0x3F, 0x41)
+			add(checkAllTemplate)
+			add(uncheckAllTemplate)
+			add(deleteSelectedTemplate)
+		})
 		add(JScrollPane(
 			templatesPanel,
 			JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
-			JScrollPane.HORIZONTAL_SCROLLBAR_NEVER))
+			JScrollPane.HORIZONTAL_SCROLLBAR_NEVER).apply {
+			minimumSize = Dimension(950, 400)
+			preferredSize = Dimension(950, 400)
+			maximumSize = Dimension(950, 400)
+		})
 
 		add(rootTemplateEditPanel)
-		val newTemplate = JButton("New").apply {
+
+		val importSettingsTemplates = JButton("Import From Settings").apply {
 			isOpaque = true
 			val currentHeight = height
 			val currentWidth = width
-			minimumSize = Dimension(currentWidth + 150, currentHeight + 40)
-			preferredSize = Dimension(currentWidth + 150, currentHeight + 40)
-			maximumSize = Dimension(currentWidth + 150, currentHeight + 40)
-			toolTipText = "Create a new template"
+			minimumSize = Dimension(currentWidth + 165, currentHeight + 40)
+			preferredSize = Dimension(currentWidth + 165, currentHeight + 40)
+			maximumSize = Dimension(currentWidth + 165, currentHeight + 40)
+			toolTipText = "Import expansion templates from settings file"
+			addActionListener {
+				JFileChooser().apply chooser@ {
+					dialogTitle = "Select Settings File to Import From"
+					fileSelectionMode = JFileChooser.FILES_ONLY
+					fileFilter = FileNameExtensionFilter("*.json", "json")
+					addChoosableFileFilter(
+						object : FileFilter()
+						{
+							override fun getDescription(): String
+							{
+								return "Settings File (*.json)"
+							}
+
+							override fun accept(f: File?): Boolean
```

Comment:
Ah, see my earlier comment.  Maybe revisit every place that matches `accept\(f: File\?`, and leave off the `?` for simplicity  It looks like there are 6 places.

Reply:
I updated it everywhere.